scipy/stats/_wilcoxon.py

```diff
@@ -0,0 +1,222 @@
+import warnings
+import numpy as np
+
+from scipy import stats
+from ._stats_py import _normtest_finish, _rankdata
+from . import _morestats
+from ._axis_nan_policy import _broadcast_arrays
+from ._hypotests import _get_wilcoxon_distr
+from scipy._lib._util import _lazywhere
+
+
+class WilcoxonDistribution:
+
+    def __init__(self, n):
+        n = np.asarray(n).astype(int, copy=False)
+        self.n = n
+        self._dists = {ni: _get_wilcoxon_distr(ni) for ni in np.unique(n)}
+
+    def _cdf1(self, k, n):
+        pmfs = self._dists[n]
+        return pmfs[:k + 1].sum()
+
+    def _cdf(self, k, n):
+        return np.vectorize(self._cdf1, otypes=[float])(k, n)
+
+    def _sf1(self, k, n):
+        pmfs = self._dists[n]
+        return pmfs[k:].sum()
+
+    def _sf(self, k, n):
+        return np.vectorize(self._sf1, otypes=[float])(k, n)
+
+    def mean(self):
+        return self.n * (self.n + 1) // 4
```

Comment:
is // correct? In the test, the mean is computed as `count * (count + 1.) * 0.25`

Reply:
It is not the correct mean for `n=1` and `n=2`. Because mean `mn` is only used to compare `k <= mn` for integer `k', it doesn't affect results, but I will change it anyway. 

I don't remember offhand whether ties or zeros can produce non-integral `k`. This null distribution is not exact anyway when the are ties or zeros, but if so, there might be something better than converting `k` to an integer by truncating it. I'll double check that.